crates/shared/src/http_solver/model.rs

```diff
@@ -213,6 +219,42 @@ impl UpdatedAmmModel {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq)]
+#[serde(untagged)]
+pub enum ExecutionPlan {
+    /// The coordinates at which the interaction should be included within a
+    /// settlement.
+    Coordinates(ExecutionPlanCoordinatesModel),
+
+    /// The interacetion should **not** be included in the settlement as
+    /// internal buffers will be used instead.
+    #[serde(with = "execution_plan_internal")]
+    Internal,
```

Comment:
Just one thought:
maybe, we might want to simulate the transactions with the "later to be internalized AMM interactions", in order to see that the internalization was legit at some point.
If we want to do that, I think its better to also ask the solvers for the executionplancoordinates even for internalized AMMs.
But given that the game of the rules are not yet finalized, I dont' know how important a possible simulation really is.

Reply:
Hmm, good point.